File: src/main/java/gg/ethereallabs/blockChess/data/PlayerData.java
package gg.ethereallabs.blockChess.data;

import java.time.Instant;

public class PlayerData {
    public int rating;
    public int gamesPlayed;
    public int wins;
    public int losses;
    public int draws;
    public Instant lastPlayed;

    public PlayerData() {
        this.rating = 800;
        this.gamesPlayed = 0;
        this.wins = 0;
        this.losses = 0;
        this.draws = 0;
        this.lastPlayed = null;
    }

    public PlayerData(int rating, int gamesPlayed) {
        this();
        this.rating = rating;
        this.gamesPlayed = gamesPlayed;
    }
}

File: src/main/kotlin/gg/ethereallabs/blockChess/data/PlayerData.kt
package gg.ethereallabs.blockChess.data

import java.time.Instant

data class PlayerData(
    var rating: Int = 800,
    var gamesPlayed: Int = 0,
    var wins: Int = 0,
    var losses: Int = 0,
    var draws: Int = 0,
    var lastPlayed : Instant? = null
)